sdk/storage_blobs/src/blob/operations/put_append_blob.rs

```diff
@@ -31,6 +32,7 @@ impl PutAppendBlobBuilder {
                     headers.add(m);
                 }
             }
+            headers.add(self.if_match);
```

Comment:
We tend to do `If-Match` and `If-None-Match` headers in a pair. Could you add support for `If-None-Match` as well?

Reply:
I believe these are handled by the match condition meaning the following fragment from the test generates if-none-match header:

```
    let result = blob
        .put_append_blob()
        .content_type("text/plain")
        .if_match(IfMatchCondition::NotMatch("*".into()))
        .await
        .unwrap_err();
```

Reply:
I guess you mean if-modified-since.   Added.

Reply:
I meant that header, but didn't realize that's how it was being handled. If that's the case, then this code looked fine.